testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -2359,6 +2369,22 @@ protected String errorMessageForInsertIntoNotNullColumn(String columnName)
         throw new UnsupportedOperationException("This method should be overridden");
     }
 
+    @Test
+    public void testInsertInTransaction()
+    {
+        skipTestUnless(hasBehavior(SUPPORTS_INSERT));
+        skipTestUnless(hasBehavior(SUPPORTS_MULTI_STATEMENT_WRITES)); // covered by testWriteNotAllowedInTransaction
+
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_tx_insert",
+                "(a bigint)")) {
+            String tableName = table.getName();
+            inTransaction(session -> assertUpdate(session, "INSERT INTO " + tableName + " VALUES 42", 1));
```

Comment:
just to make sure, this would work without your change(removing that 'verify')  right?

Reply:
for Detla, the test would break at the `skipTestUnless(hasBehavior(SUPPORTS_MULTI_STATEMENT_WRITES));` because Delta doesn't support this. Only Hive does ATM, AFAIR.